source/grid/manifold_lib.cc

```diff
@@ -1374,7 +1464,29 @@ ::compute_chart_points (const ArrayView<const Point<spacedim>> &surrounding_poin
       bool inside_unit_cell = true;
       for (unsigned int i=0; i<surrounding_points.size(); ++i)
         {
-          chart_points[i] = pull_back(cell, surrounding_points[i]);
+          // some initial guesses - assuming that the chart points end up in a
+          // regular (cube-like) pattern which they often do).
+
+          // if we have already computed three points, we can guess the fourth
+          // to be the missing corner point of a rectangle
+          if (i == 3)
+            {
+              const Point<dim> p3 = chart_points[1] +
+                                    Point<dim>(chart_points[2]-chart_points[0]);
```

Comment:
pedantic: the difference type here is `Tensor<1, dim>`

Reply:
I now, but I want to get a position (thus `Point`) in the reference coordinate system of the chart. Or do I misunderstand your comment?

Reply:
My point is that `Point<dim>::operator+` expects a `Tensor<1,dim>`, which is already the type of the difference `chart_points[2] - chart_points[0]`: we don't need the extra `Point<dim>(...)` call in that line.

Reply:
Oh yes, I didn't see that. Thanks for pointing it out.